### ISSUE TITLE
cice.setup for tests : --bgen default does not work

### ISSUE THREAD

**Issue by phil-blain** [CICE-Consortium/CICE#301]
When using cice.setup to create an individual test (cice.setup --test), using the` --bgen default` option does not work.  The code in cice.setup that generate the baseline name based on the hash and date is only executed for test suites, and not for single tests.  Since the `default` option is mentioned for individual tests [in the documentation](https://cice-consortium-cice.readthedocs.io/en/master/user_guide/ug_testing.html#individual-tests), I think this is a bug.

**Comment by apcraig** [comment]
@phil-blain, thanks for finding the issues and reporting them.  When you say it doesn't work, I assume it creates a directory called "default" and places the results there?  The bgen and bcmp work for single tests, it sounds like it's just using the special value of "default" that doesn't work.  Do I understand correctly?  I will fix this, should be fairly straight-forward.

**Comment by phil-blain** [comment]
Yes, it is only the 'default' special value that does not work for bgen. It works for bcmp. What happens is cice.setup outputs the following:
``` 
./cice.setup:
ICE_SANDBOX  = /local/drive1/cmdd/afsgphb/Code/cice.phil
ICE_CASENAME = cesium_intel_smoke_gx3_1x1_diag1.new-test
ICE_CASEDIR  = /local/drive1/cmdd/afsgphb/CICEDIRS/tests/default-testing/cesium_intel_smoke_gx3_1x1_diag1.new-test
ICE_MACHINE  = cesium
ICE_COMPILER = intel
ICE_PES      = 1x1
ICE_GRID     = gx3 (100x116) blocksize=100x116x1
ICE_DECOMP   = cartesian slenderX2
cdate: Undefined variable.
ctime: Undefined variable.
shhash: Undefined variable.
```

It then errors at line 661 : `set baseGen = cice.${shhash}.${d1}-${t1}` because shhash is not defined (the code that defines it at line 388 is only executed if creating a suite).
The test directory is correctly created, but almost all values in cice.settings are "undefined" and the cice.test, cice.submit scripts are missing (since cice.setup stops before creating them).

**Comment by apcraig** [comment]
This is being fixed in #300.  the default bgen and bcmp settings will work with a single test.

**Comment by apcraig** [comment]
PR #300 complete
